sandbox_osteinnes/MarioBros_libGDX/core/src/com/mygdx/testgamev1/Sprites/Brick.java

```diff
@@ -5,29 +5,50 @@
         Link to the tutorial playlist: https://www.youtube.com/playlist?list=PLZm85UZQLd2SXQzsF-a0-pPF6IWDDdrXt
  */
 
-import com.badlogic.gdx.maps.tiled.TiledMap;
-import com.badlogic.gdx.math.Rectangle;
-import com.badlogic.gdx.physics.box2d.World;
+import com.badlogic.gdx.Gdx;
+import com.mygdx.testgamev1.Scenes.Hud;
+import com.mygdx.testgamev1.Screens.PlayScreen;
 
 /**
- * The Brick-class extends InteractiveTileObjects. This is because the Brick and Coin objects
+ * The Brick-class extends InteractiveTileObject. This is because the Brick and Coin objects
  * will both behave similarly. Therefore their common traits can be written in a parent-class (r: super)
  *
  * @author Ole-martin Steinnes
  * @version 1
  */
-public class Brick extends InteractiveTileObjects {
+public class Brick extends InteractiveTileObject {
 
     /**
      * Constructor in the Brick-class. Takes in fields for creation of defined Box2D-boxes.
      * There will also be implemented responses to interactions with the player, in the future.
      *
-     * @param world World of the box2d restraints
-     * @param map Map of the game.
-     * @param bounds Bounds of the map.
-     * @param pixelsPerMeter Pixels Per Meter unit from the game settings.
+     * @param playScreen Screen the world is attached to.
      */
-    public Brick(World world, TiledMap map, Rectangle bounds, float pixelsPerMeter) {
-        super(world, map, bounds, pixelsPerMeter);
+    public Brick(PlayScreen playScreen) {
+        super(playScreen);
+
+    }
+
+    @Override
+    public void initObject() {
+        fixture.setUserData(this);
+        isActive = true;
+        setCategoryFilter(playScreen.getGame().getBrickBit());
+    }
+
+    @Override
+    public void onHeadHit() {
+        Gdx.app.log("Brick", "Collision");
+        setCategoryFilter(playScreen.getGame().getDestroyedBit());
+
+        // Removes the brick texture.
+        getCell().setTile(null);
+        if (isActive) {
+            Hud hud = playScreen.getHud();
+
+            hud.addScore(200);
```

Comment:
Score is stored in the HUD? I would guess it would be better to store game state somewhere else and just let HUD display the necessary info.